flowetl/etl/tests/test_dummy_callables.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from unittest.mock import patch, MagicMock, PropertyMock
+from etl.dummy_task_callables import dummy_callable, dummy_failing_callable
+
+
+def test_dummy_callable_succeeds_with_no_TASK_TO_FAIL_env_var_set():
+    """
+    Test that when no TASK_TO_FAIL env var set the dummy callable
+    succeeds.
+    """
+    dag_run = {}
+    task_instance = MagicMock()
```

Comment:
It's fine to use a mock here, but since all you need is the `task_id` to be present, you could also define a `DummyTask` class which provides a stub implementation:
```python
class DummyTask:
    def __init__(self, task_id):
        self.task_id = task_id
```
and then use this in the test as:
```python
task_instance = DummyTask(task_id="init")
```
I tend to find this a little bit more readable, and also it makes it a bit more obvious that the only thing your tests rely on is that the `task_id` attribute is present and has a certain value. (Wheras a mock object is typically used when you need to verify an expected method call happened, rather than just providing stubbed out behaviour).

Reply:
cool changed!